libs/domain/cart/src/guards/cart-price.guard.ts

```diff
@@ -0,0 +1,22 @@
+import { inject } from '@spryker-oryx/di';
+import { PriceModeChangeGuard } from '@spryker-oryx/site';
+import { Observable } from 'rxjs';
+import { CartService } from '../services';
+
+export interface Guard {
+  isAllowed(): Observable<boolean>;
+}
+
+export class CartPriceModeChangeGuard implements Guard {
+  constructor(protected cartService = inject(CartService)) {}
+
+  isAllowed(): Observable<boolean> {
+    return this.cartService.isEmpty();
+  }
+}
```

Comment:
Do we really need this guard, why we can't just check `this.cartService.isEmpty();` inside component?

Reply:
Yes we need this guard from cart domain, to check  in site domain for price mode selection if there is a empty cart.
And we can’t import cart service on site domain because of circular dependency.

Reply:
Why we have `interface Guard ` there then?